mne/io/curry/curry.py

```diff
@@ -0,0 +1,238 @@
+    sfreq = float(param_dict["samplefreqhz"])
+    offset = float(param_dict["triggeroffsetusec"]) * CAL
+    time_step = float(param_dict["sampletimeusec"]) * CAL
+    data_format = param_dict["dataformat"]
+
+    if (sfreq == 0) and (time_step != 0):
+        sfreq = 1. / time_step
+
+    #####################################
+    # read labels from label files
+
+    ch_names = []
+    ch_pos = []
+
+    save_labels = False
+    save_ch_pos = False
+    with open(fname_base + LABEL_FILE_EXTENSION[curry_vers]) as fid:
+        for line in fid:
+
+            if re.match("LABELS.*? END_LIST", line):
+                save_labels = False
+
+            # if "SENSORS END_LIST" in line:
+            if re.match("SENSORS.*? END_LIST", line):
+                save_ch_pos = False
+
+            if save_labels and line != "\n":
+                    ch_names.append(line.replace("\n", ""))
+
+            if save_ch_pos:
+                ch_pos.append(line.split("\t"))
+
+            if re.match("LABELS.*? START_LIST", line):
+                save_labels = True
+
+            # if "SENSORS START_LIST" in line:
+            if re.match("SENSORS.*? START_LIST", line):
+                save_ch_pos = True
+
+    ch_pos = np.array(ch_pos, dtype=float)
+    # TODO find a good method to set montage (do it in read_montage instead?)
+
+    info = create_info(ch_names, sfreq)
+
+    # TODO; There's still a lot more information that can be brought into info["chs"]. However i'm not sure what to do with MEG chans here
+    for ch_dict in info["chs"]:
+        ch_dict["cal"] = CAL
+
+    return info, n_trials, n_samples, curry_vers, data_format
+
+
+def read_raw_curry(input_fname, preload=False):
+    """
+    Read raw data from Curry files.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory). If the curry file
+        is stored in ASCII data format, then preload must be `True`.
+
+    Returns
+    -------
+    raw : instance of RawCurry
+        A Raw object containing CURRY data.
+
+    """
+
+    DATA_FILE_EXTENSION = {7: '.dat', 8: '.cdt'}
+
+    # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+    fname_base, ext = input_fname.split(".", maxsplit=1)
+
+    curry_vers = _get_curry_version(ext)
+    _check_missing_files(input_fname, fname_base, curry_vers)
+
+    info, n_trials, n_samples, curry_vers, data_format = _read_curry_info(fname_base, curry_vers)
+    events = _read_curry_events(fname_base, curry_vers)
```

Comment:
Now that I've read the files better, these are not events but annotations. We should expose `read_annotations_curry`